filecoin-proofs/src/api.rs

```diff
@@ -481,3 +589,121 @@ fn pad_safe_fr(unpadded: &FrSafe) -> Fr32Ary {
     res[0..31].copy_from_slice(unpadded);
     res
 }
+
+#[test]
+fn test_generate_piece_commitment() -> Result<(), failure::Error> {
+    fn blah(data: &[u8]) -> Result<Commitment, failure::Error> {
+        let mut file = NamedTempFile::new().expects("could not create named temp file");
+        file.write_all(data)?;
+        let (comm_p, _) =
+            generate_piece_commitment(file.path(), UnpaddedBytesAmount(data.len() as u64))?;
+        Ok(comm_p)
+    }
+
+    {
+        // this scope tests comm_p generation for all byte lengths up to the minimum piece
+        // alignment when writing a piece to a sector
+        let max = 127;
```

Comment:
TODO: Factor this magical number into a constant somewhere and give that constant a useful name.